File: src/main/java/net/cubizor/cubiloc/locale/DefaultLocaleProvider.java
package net.cubizor.cubiloc.locale;

import java.util.Locale;

/**
 * Default locale provider that handles common types.
 * Supports String locale codes and java.util.Locale objects.
 */
public class DefaultLocaleProvider implements LocaleProvider<Object> {
    
    private final Locale fallbackLocale;
    
    public DefaultLocaleProvider() {
        this.fallbackLocale = Locale.forLanguageTag("en-US");
    }
    
    public DefaultLocaleProvider(Locale fallbackLocale) {
        this.fallbackLocale = fallbackLocale;
    }
    
    @Override
    public boolean supports(Class<?> type) {
        return String.class.isAssignableFrom(type) 
            || Locale.class.isAssignableFrom(type);
    }
    
    @Override
    public Locale getLocale(Object entity) {
        if (entity == null) {
            return fallbackLocale;
        }
        
        if (entity instanceof Locale) {
            return (Locale) entity;
        }
        
        if (entity instanceof String) {
            String localeStr = (String) entity;
            // Handle both "en_US" and "en-US" formats
            localeStr = localeStr.replace("_", "-");
            return Locale.forLanguageTag(localeStr);
        }
        
        return fallbackLocale;
    }
}

File: src/main/kotlin/net/cubizor/cubiloc/locale/DefaultLocaleProvider.kt
package net.cubizor.cubiloc.locale

import java.util.Locale

class DefaultLocaleProvider(
    private val fallbackLocale: Locale = Locale.ENGLISH,
) : LocaleProvider<Any> {

    override fun supports(type: Class<*>): Boolean =
        String::class.java.isAssignableFrom(type) || Locale::class.java.isAssignableFrom(type)

    override fun getLocale(entity: Any): Locale? = when (entity) {
        is Locale -> entity
        is String -> Locale.forLanguageTag(entity.replace("_", "-"))
        else -> fallbackLocale
    }
}
